src/utils/leap.ts

```diff
@@ -0,0 +1,109 @@
+import {
+  OfflineSigner,
+  LikeCoinWalletConnectorInitResponse,
+  LikeCoinWalletConnectorOptions,
+} from '../types';
+
+export async function initLeap(
+  options: LikeCoinWalletConnectorOptions,
+  trys = 0
+): Promise<LikeCoinWalletConnectorInitResponse> {
+  if (!window.leap || !window.getOfflineSignerAuto) {
+    if (trys < options.initAttemptCount) {
+      await new Promise(resolve => setTimeout(resolve, 1000));
+      return initLeap(options, trys + 1);
+    }
+    throw new Error('LEAP_NOT_INSTALLED');
+  }
+
+  try {
+    // Some Leap configs not support empty string
+    const coinGeckoId = options.coinGeckoId || undefined;
+    const walletUrlForStaking = options.walletURLForStaking || undefined;
+
+    await window.leap.experimentalSuggestChain({
+      chainId: options.chainId,
+      chainName: options.chainName,
+      rpc: options.rpcURL,
+      rest: options.restURL,
+      stakeCurrency: {
+        coinDenom: options.coinDenom,
+        coinMinimalDenom: options.coinMinimalDenom,
+        coinDecimals: options.coinDecimals,
+        coinGeckoId,
+      },
+      walletUrlForStaking,
+      bip44: {
+        coinType: options.coinType,
+      },
+      bech32Config: {
+        bech32PrefixAccAddr: options.bech32PrefixAccAddr,
+        bech32PrefixAccPub: options.bech32PrefixAccPub,
+        bech32PrefixValAddr: options.bech32PrefixValAddr,
+        bech32PrefixValPub: options.bech32PrefixValPub,
+        bech32PrefixConsAddr: options.bech32PrefixConsAddr,
+        bech32PrefixConsPub: options.bech32PrefixConsPub,
+      },
+      currencies: [
+        {
+          coinDenom: options.coinDenom,
+          coinMinimalDenom: options.coinMinimalDenom,
+          coinDecimals: options.coinDecimals,
+          coinGeckoId,
+        },
+      ],
+      feeCurrencies: [
+        {
+          coinDenom: options.coinDenom,
+          coinMinimalDenom: options.coinMinimalDenom,
+          coinDecimals: options.coinDecimals,
+          coinGeckoId,
+          gasPriceStep: {
+            low: options.gasPriceStepLow,
+            average: options.gasPriceStepAverage,
+            high: options.gasPriceStepHigh,
+          },
+        },
+      ],
+      coinType: options.coinType,
+      features: ['ibc-go', 'ibc-transfer', 'no-legacy-stdTx', 'stargate'],
```

Comment:
Does leap has a list of supported features? Many of these are marked deprecated for keplr

Reply:
Updated with `ibc-go` and `ibc-transfer` only

Ref: https://docs.leapwallet.io/cosmos/connect-to-leap/suggest-chain